augur/distance.py

```
@@ -73,54 +366,99 @@ def run(args):
+
+    final_distances_by_node = {}
+    distance_map_names = []
+    for compare_to, attribute, distance_map_file in zip(args.compare_to, args.attribute_name, args.map):
+        # Load the given distance map.
+        distance_map = read_distance_map(distance_map_file)
+        distance_map_names.append(distance_map.get("name", distance_map_file))
 
-        # Calculate the distance for this node for each requested mask and store
-        # the result in the corresponding attribute name.
-        distances_by_node[node] = {}
-        for mask_name, attribute in attributes_by_mask.items():
-            distances_by_node[node][attribute] = mask_distance(
-                root_node_sequence,
-                node_sequence,
-                masks[mask_name]
+        # Use the distance map to calculate distances between all samples in the
+        # given tree and the desired target(s).
+        if compare_to == "root":
+            # Calculate distance between the root and all samples.
+            distances_by_node = get_distances_to_root(
+                tree,
+                sequences_by_node_and_gene,
+                distance_map
             )
+        elif compare_to == "ancestor":
+            # Calculate distance between the last ancestor for each sample in a
+            # previous season.
+            distances_by_node = get_distances_to_last_ancestor(
+                tree,
+                sequences_by_node_and_gene,
+                distance_map,
+                latest_date
+            )
+        elif compare_to == "pairwise":
+            # Calculate distance between each sample and all other samples in a
+            # previous season.
+            distances_by_node = get_distances_to_all_pairs(
+                tree,
+                sequences_by_node_and_gene,
+                distance_map,
+                earliest_date,
+                latest_date
+            )
+        else:
+            pass
```

I'd argue this should raise an error instead of silently not doing anything.